readme: update and split into various parts (#289)
fort-nix/nix-bitcoin: README.md

        <img src="https://img.shields.io/github/contributors-anon/fort-nix/nix-bitcoin" alt="GitHub contributors">
    </a>
    <a href="https://github.com/fort-nix/nix-bitcoin/releases" target="_blank">
        <img src="https://img.shields.io/github/downloads/fort-nix/nix-bitcoin/total" alt="GitHub downloads">
    </a>
</p>
<br/>

nix-bitcoin is a collection of Nix packages and NixOS modules for easily installing **full-featured Bitcoin nodes** with an emphasis on **security**.

Overview
---
A Bitcoin node verifies the Bitcoin protocol and provides ways of interacting with the Bitcoin network. nix-bitcoin
nodes are used for a variety of purposes and can serve as personal or merchant wallets, second layer public
infrastructure and as backends for Bitcoin applications. In all cases, the aim is to provide security and privacy by
default. However, while nix-bitcoin is used in production today, it is still considered experimental.

A full installation of nix-bitcoin is usually deployed either on a dedicated (virtual) machine or runs in a container
and is online 24/7. Alternatively, the Nix packages, NixOS modules and configurations can be used independently and
combined freely.

> nix-bitcoin is built on top of Nix and NixOS because that allows using abstractions powerful enough to keep it highly
> customizable and maintainable. Testament to this are nix-bitcoin's robust security features and its potent test
> framework.  However, running nix-bitcoin does not require any previous experience with the Nix ecosystem.

Reviewer: Nit, make wording more concise and poignant:
```suggestion
nix-bitcoin is built on top of Nix and NixOS which provide powerful abstractions to keep it highly
customizable and maintainable. Testament to this are nix-bitcoin's robust security features and its potent test
framework.  However, running nix-bitcoin does not require any previous experience with the Nix ecosystem.
```

Author: Fixed in bcedf69549652b9d201cc9e02f64c086d6501d14

        <img src="https://img.shields.io/github/v/release/fort-nix/nix-bitcoin" alt="GitHub tag (latest SemVer)">
    </a>
    <a href="https://cirrus-ci.com/github/fort-nix/nix-bitcoin" target="_blank">
        <img src="https://api.cirrus-ci.com/github/fort-nix/nix-bitcoin.svg?branch=master" alt="CirrusCI actions status">
    </a>
    <a href="https://github.com/fort-nix/nix-bitcoin/commits/master" target="_blank">
        <img src="https://img.shields.io/github/commit-activity/y/fort-nix/nix-bitcoin" alt="GitHub commit activity">
    </a>
    <a href="https://github.com/fort-nix/nix-bitcoin/graphs/contributors" target="_blank">
        <img src="https://img.shields.io/github/contributors-anon/fort-nix/nix-bitcoin" alt="GitHub contributors">
    </a>
    <a href="https://github.com/fort-nix/nix-bitcoin/releases" target="_blank">
        <img src="https://img.shields.io/github/downloads/fort-nix/nix-bitcoin/total" alt="GitHub downloads">
    </a>
</p>
<br/>

nix-bitcoin is a collection of Nix packages and NixOS modules for easily installing **full-featured Bitcoin nodes** with an emphasis on **security**.

> Goals

Reviewer 2: I'd suggest to replace the whole Goals section with something like the following:

Overview
---
A Bitcoin node verifies the Bitcoin protocol and provides ways of interacting with the Bitcoin network. nix-bitcoin nodes are used for a variety of purposes and can serve as personal or merchant wallets, second layer public infrastructure and as backends for Bitcoin applications. In all cases, the aim is to provide security and privacy by default. However, while nix-bitcoin is used in production today, it is still considered experimental.

A full installation of nix-bitcoin is usually deployed either on a dedicated (virtual) machine or runs in a container and is online 24/7. Alternatively, the Nix packages, NixOS modules and configurations can be used independently and combined freely.

nix-bitcoin is built on top of Nix and NixOS because that allows using abstractions powerful enough to keep it highly customizable and maintainable. Testament to this are nix-bitcoin's security features and its potent test framework. However, running nix-bitcoin does not require any previous experience with the nix ecosystem.

---
* Make it easy to deploy a reasonably secure Bitcoin node with a usable wallet
* Allow providing public infrastructure for Bitcoin and higher-layer protocols
* Be a usable, reasonably secure platform for trustless Bitcoin yield generation
* Be a reproducible and extensible platform for applications building on Bitcoin

Features
---
Default
* bitcoind with outbound connections through Tor and inbound connections through an onion service. By default loaded with banlist of spy nodes.
* [clightning](https://github.com/ElementsProject/lightning) with outbound connections through Tor, onion service unannounced
* includes "nodeinfo" script which prints basic info about the node
* adds non-root user "operator" who has access to client tools (ex. bitcoin-cli, lightning-cli)

In `configuration.nix` the user can enable

> * announcing a clightning onion service and [clightning plugins](https://github.com/lightningd/plugins):

Reviewer 2: How about linking to the plugin section in usage.md here? Would save some space. Perhaps we can keep some highlights here, like clboss.

Author: I like seeing all the c-lightning plugins. We can remove this section if we add more plugins and it takes up too much space.
